internal/database/mariadb/issue_match.go

```diff
@@ -177,48 +269,68 @@ func (s *SqlDatabase) buildIssueMatchStatement(baseQuery string, filter *entity.
 	filterParameters = buildQueryParameters(filterParameters, filter.IssueType)
 	filterParameters = buildQueryParametersCount(filterParameters, filter.Search, wildCardFilterParamCount)
 
-	if withCursor {
-		filterParameters = append(filterParameters, cursor.Value)
+	if withOrder {
+		filterParameters = append(filterParameters, cursor.ParameterValues...)
+	}
+
+	if withLimit {
 		filterParameters = append(filterParameters, cursor.Limit)
 	}
 
 	return stmt, filterParameters, nil
 }
 
-func (s *SqlDatabase) GetAllIssueMatchIds(filter *entity.IssueMatchFilter) ([]int64, error) {
+func (s *SqlDatabase) GetAllIssueMatchCursors(filter *entity.IssueMatchFilter, order []entity.Order) ([]string, error) {
```

Comment:
Why do we this method actually? How does the logic differ from [ListIssues](https://github.com/cloudoperators/heureka/blob/drochow/issue-310/issue_match_order/internal/app/issue_match/issue_match_handler.go#L69) ?